1.1 updates - Media Explorer library added for adding videos to posts

nimmolo opened the pull request:
See commit notes, please test!

nimmolo committed:
Merge remote-tracking branch 'silviapfeiffer/master'

nimmolo committed:
1.1 updates

Bug fixes for formatting of $zero_message if no videos found, video
categories appended to post

Reduce use of more costly require_once in init, autoload hosts from glob

Big update: Media Explorer project from WP.com incorporated into
plugin, as a framework for extending Media Uploader backbone.js. Checks
to be sure class is not already instantiated. New ev functions extend
this class to find EV videos in the media uploader box. Search of
videos enabled via js.

Big difference from Media Explorer as it is usually used: we don’t use
it to make AJAX calls to the services, since we already do that to
import the videos. We just query the external videos post type and get
the video data MEXP needs out of the post and post_meta.

Uninstall.php file added for cleanup

nimmolo committed:
Screenshot for media uploader

Screenshot explanation also added to readme

nimmolo committed:
Version number

whoops

nimmolo committed:
Updated description

in readme and in main plugin file

silviapfeiffer commented on external-videos.php:
```diff
@@ -47,17 +47,17 @@ public function __construct() {
-    require_once( plugin_dir_path( __FILE__ ) . 'hosts/ev-dotsub.php' );
-    require_once( plugin_dir_path( __FILE__ ) . 'hosts/ev-wistia.php' );
-    require_once( plugin_dir_path( __FILE__ ) . 'hosts/ev-dailymotion.php' );
+    require( plugin_dir_path( __FILE__ ) . 'core/ev-admin.php' );
```
Why did you replace the require_once includes?

nimmolo replied:
Researching the plugin i came across arguments that require_once required significantly more work from the server, and if the code has no chance of double includes, it's always better to require. Now that i'm looking it up again, seems it's not SUCH a big difference. http://stackoverflow.com/questions/186338/why-is-require-once-so-bad-to-use

silviapfeiffer replied:
Ah ok, I was just curious. Whatever works...

silviapfeiffer commented on hosts/dailymotion/mexp-ev-dailymotion-service.php:
```diff
@@ -0,0 +1,136 @@
+
+    // Set up items
+		foreach ( $query_response as $index => $video ) {
+      setup_postdata( $video );
```
Indent seems a bit broken

nimmolo replied:
Yes this is weird! I have my editor set to do double spaces, no tabs, and everything seemed ok at commit time.

silviapfeiffer commented on hosts/youtube/mexp-ev-youtube-service.php:
```diff
@@ -0,0 +1,135 @@
+			return false;
+
+    // Set up items
+		foreach ( $query_response as $index => $video ) {
```
more weird indents - sorry about being pedantic

nimmolo replied:
Fixed all these.

silviapfeiffer reviewed (commented):
Sorry, being a bit pedantic, but some of the code is inconsistently indented. I'll try it out now.

silviapfeiffer commented:
Not your fault, but I'm getting:
`Warning: dirname() expects exactly 1 parameter, 2 given in .../wordpress/external-videos/mexp/media-explorer.php on line 20`

Note: I'm on php 5.6.16


You will want to replace 'dirname( __FILE__,2) ' with 'dirname(dirname( __FILE__)) ' - it's the easiest.

silviapfeiffer commented:
BTW: I also noticed that https://wordpress.org/plugins/external-videos/screenshots/ isn't working. I think we need to add the directory `assets` to the readme file for the screenshots

nimmolo commented:
oh damn! didn't notice that dirname parameter was php 7-only.

nimmolo commented:
I looked through all the "Video" plugins currently on the WP repo, just to check out what's out there.

In light of that, and the fact that video embeds in WP are easier than ever, I'd like to propose a new tagline and description for what the plugin does, that more clearly differentiates it from others. Feel free to edit or discard.

nimmolo committed:
Description changes and requested fixes

Indents fixed in media explorer code
Directory added for screenshots
Proposed new short and long plugin descriptions to better situate
plugin in repository
PHP 7 dirname use rewritten

silviapfeiffer commented:
I like the new tagline! 😀

nimmolo commented:
Maybe we should add the word "sync" to the tags. The tags didnt update on SVN last time, did i do them right?

For the screenshots, I checked other plugins in SVN and it seems they shouldn't need the /assets directory folder listing, or even any filename at all, as long as they are sequentially numbered. 

Check here: https://plugins.svn.wordpress.org/wordpress-seo/trunk/readme.txt

I wonder if, on SVN, /assets maybe has to be synced separately, since it's outside of /trunk. Let me know if you want me to look into this further.

nimmolo committed:
Screenshot list revisions

Filenames apparently not needed, as long as they are named
“screenshot-#”

silviapfeiffer commented:
Looking good. I'm landing and releasing it. Let's see if we get the tags updated.

silviapfeiffer commented:
OK, released. The tags appeared, but they get ordered alphabetically.
Congratulations again! :-)

nimmolo commented:
Thanks Silvia!

From [this newer page on the repo](https://developer.wordpress.org/plugins/wordpress-org/plugin-assets/) i'm reading the assets folder does have to be uploaded separately. I think that's why the icon and screenshots haven't been working. I'll see if i can fix it but i may not have direct access to SVN

nimmolo commented:
Just checked - i don't have access to SVN as committer, but basically we just have to create an /assets top level folder separately from /branches, /tags, and /trunk.

![screenshot](https://cloud.githubusercontent.com/assets/1948095/22265395/c11ed00e-e230-11e6-945b-2adc640e2fe7.png)
